### PR TITLE
Fix the test framework is failing in CRLF env.

### LINKED ISSUE #1318: Tests fail on Windows with panic in tests/system.rs
On Windows, pretty much every test fails with the following panic:

```
running 9 tests
test format_lines_errors_are_reported ... ok
test rustfmt_diff_make_diff_tests ... ok
test rustfmt_diff_no_diff_test ... ok
test checkstyle_test ... ok
test stdin_formatting_smoke_test ... ok
test idempotence_tests ... FAILED
test coverage_tests ... FAILED
test system_tests ... FAILED
test self_tests ... FAILED

thread 'self_tests' panicked at 'Empty diff? Maybe due to a missing a newline at the end of a file?', tests\system.rs:323
stack backtrace:
   0:     0x7ff6ca1253f8 - std::panicking::default_hook::{{closure}}
                        at C:\bot\slave\nightly-dist-rustc-win-msvc-64\build\src\libstd\panicking.rs:356
   1:     0x7ff6ca124882 - std::panicking::default_hook
                        at C:\bot\slave\nightly-dist-rustc-win-msvc-64\build\src\libstd\panicking.rs:361
   2:     0x7ff6ca1282bd - std::panicking::rust_panic_with_hook
                        at C:\bot\slave\nightly-dist-rustc-win-msvc-64\build\src\libstd\panicking.rs:545
   3:     0x7ff6c99530d7 - std::panicking::begin_panic<&str>
                        at C:\bot\slave\nightly-dist-rustc-win-msvc-64\build\src\libstd\panicking.rs:507
   4:     0x7ff6c99a2124 - system::handle_result
                        at C:\Users\Marcus\Source\rustfmt\tests\system.rs:323
   5:     0x7ff6c99a0bc4 - system::idempotent_check
                        at C:\Users\Marcus\Source\rustfmt\tests\system.rs:256
   6:     0x7ff6c999ec74 - system::check_files<core::iter::Chain<core::iter::Chain<core::iter::Map<core::iter::Chain<std::fs::ReadDir, std::fs::Read
Dir>, fn(core::result::Result<std::fs::DirEntry, std::io::error::Error>) -> collections::string::String>, core::option::IntoIter<collections::string
::String>>, core::option::IntoIter<collections::string::String>>>
                        at C:\Users\Marcus\Source\rustfmt\tests\system.rs:181
   7:     0x7ff6c999daf0 - system::self_tests
                        at C:\Users\Marcus\Source\rustfmt\tests\system.rs:126
   8:     0x7ff6c9fc835a - test::{{impl}}::call_box<(),closure>
```

The problem is located [here](https://github.com/rust-lang-nursery/rustfmt/blob/master/tests/system.rs#L321):
```rust
        if fmt_text != text {
            let diff = make_diff(&text, &fmt_text, DIFF_CONTEXT_SIZE);
            assert!(!diff.is_empty(),
                    "Empty diff? Maybe due to a missing a newline at the end of a file?");
            failures.insert(file_name, diff);
        }
```

Basically, `fmt_text` has new-line characters unified to `\n`, while `text` still has Windows' style `\r\n` line-endings. So, `fmt_text != text` evaluates to true, however, `make_diff` properly accounts for the line-ending differences so it returns an empty diff. 

This could likely be fixed by simply applying a string replace of `\r\n` to `\n` on `text` (and possibly `fmt_text`, too, just to be safe). 

If I find some free-time soon, I'll try fixing this myself, if no one else gets around to it.

### PR DESCRIPTION
**Pull request by wada314**
Fixes #1318 .
This just fixes the framework error: Many tests are still failing under Windows machine because of CRLF newline. Maybe I follow up with another pull requests.

Note: I cannot use str::lines() method to compare because that method does not respects the last empty line in the file.
https://doc.rust-lang.org/std/string/struct.String.html#method.lines

### CODE REVIEW

**Comment by wada314** [comment]
Thank you for reviewing! 

Yes I agree for your concern. I wrote a function to compare bytes automatically converting `\r\n` to `\n`, without re-allocating I believe. Could you please take another look?

File: tests/system.rs [hunk 1]
```diff
@@ -431,3 +433,33 @@ fn rustfmt_diff_no_diff_test() {
...
+            }
+        } else {
+            c
+        })
+    }
+}
+
+#[test]
+fn string_eq_ignore_newline_repr_test() {
+    assert!(string_eq_ignore_newline_repr("a\nb\nc\rd", "a\nb\r\nc\rd"));
```
Comment (nrc) [inline tests/system.rs]: Could you add a few more test cases please? For example, where the left or right are empty, where they match trivially, where they don't match, and where the left and right are different lengths.

File: tests/system.rs [hunk 2]
```diff
@@ -431,3 +433,33 @@ fn rustfmt_diff_no_diff_test() {
     let diff = make_diff("a\nb\nc\nd", "a\nb\nc\nd", 3);
     assert_eq!(diff, vec![]);
 }
+
+// Compare strings without distinguishing between CRLF and LF
+fn string_eq_ignore_newline_repr(left: &str, right: &str) -> bool {
+    let left = BytesIgnoreNewlineRepr(left.bytes().peekable());
```
Comment (nrc) [inline tests/system.rs]: I think it would be better to use `chars` rather than `bytes` here, and iterate over `char`s (I'm not sure if it is possible, but I feel like, there might be some subtle bug if `b'\r'` is a byte in a longer char

**Comment by wada314** [comment]
Yeah I somehow messed my repository by rebasing 😆 I needed to squash my commits to fix it, hope it doesn't matter.

> chars and bytes

Personally I lean into bytes but I don't care actually so changed to chars.
Just I didn't want to encode every bytes into char. The input bytes are utf8 so I believe it's guaranteed to be safe in < 0x7f area.

> tests

added some.

**Comment by nrc** [comment]
Great, thanks for the changes!
